### DISCUSSION TITLE
How to add procedure to the root route / home route ("/")?

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [trpc/trpc#4863] [Q&A] [answered]
```ts
export const appRouter = router({
  '/': publicProcedure.query(() => 'Hello World'),
});
```
I tried that, but return 404.

Thanks

**Comment by MariuszKogut** [comment, marked as answer]
You have to use a empty string for the root route:

```typescript
const appRouter = router({
  '': publicProcedure.query(() => 'Hello World'),
})`
```

**Reply by cibelius** [reply]
This works, but its a bit ugly to call with a caller created with [createCallerFactory()](https://trpc.io/docs/server/server-side-calls) or a client created with [createTRPCClient()](https://trpc.io/docs/client/vanilla/setup), since I need to define the empty string in the request `''`, for example:

```typescript
const createCaller = createCallerFactory(appRouter)
const client = createCaller()

await client['']()
```

preferably I'd still be able to simply do: `await client()` if I want to call the root route.
